### ISSUE TITLE
微调过程loss为0， grad_norm为nan

### ISSUE THREAD

**Issue by lychee-2724540853** [ymcui/Chinese-LLaMA-Alpaca-3#29]
### 提交前必须检查以下项目

- [X] 请确保使用的是仓库最新代码（git pull）
- [X] 已阅读[项目文档](https://github.com/ymcui/Chinese-LLaMA-Alpaca-3/wiki)和[FAQ章节](https://github.com/ymcui/Chinese-LLaMA-Alpaca-3/wiki/常见问题)并且已在Issue中对问题进行了搜索，没有找到相似问题和解决方案。
- [X] 第三方插件问题：例如[llama.cpp](https://github.com/ggerganov/llama.cpp)、[text-generation-webui](https://github.com/oobabooga/text-generation-webui)等，建议优先去对应的项目中查找解决方案。

### 问题类型

模型训练与精调

### 基础模型

Llama-3-Chinese-8B（基座模型）

### 操作系统

Linux

### 详细描述问题

在自己数据集上进行微调，但是训练过程loss为0，grad_norm为nan，请问是什么问题？

### 依赖情况（代码类问题务必提供）

```
# 请在此处粘贴依赖情况（请粘贴在本代码块里）
```

### 运行日志或截图

![image](https://github.com/ymcui/Chinese-LLaMA-Alpaca-3/assets/57849782/7f4a085b-594a-46d5-b860-05f8dbf7067a)

**Comment by iMountTai** [comment]
开启bf16了吗?

**Comment by lychee-2724540853** [comment]
是的，按照提供的参数脚本训练的 @iMountTai

**Comment by iMountTai** [comment]
确认一下数据长度和你设置的`max_seq_lenth`是否匹配，可能是因为你的output并没有计入logits

**Comment by auroralsun** [comment]
I ran into the same situation where I had limited the length of the data set to 3000 and max_length to 4096
![image](https://github.com/ymcui/Chinese-LLaMA-Alpaca-3/assets/59675069/8cd7d69a-ef8a-4f38-9af9-ffadf6ea40de)

**Comment by zzb1185** [comment]
请问解决了吗？目前我也遇到了同样的问题

**Comment by chenmonster** [comment]
使用 [kigner/ruozhiba-llama3-tt](https://huggingface.co/datasets/kigner/ruozhiba-llama3-tt) 数据集在 [hfl/llama-3-chinese-8b-instruct-v2](https://huggingface.co/hfl/llama-3-chinese-8b-instruct-v2) 上训练也出现了这个问题，暂时找到解决办法...

**Comment by auroralsun** [comment]
> I ran into the same situation where I had limited the length of the data set to 3000 and max_length to 4096 ![image](https://private-user-images.githubusercontent.com/59675069/329804812-8cd7d69a-ef8a-4f38-9af9-ffadf6ea40de.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY1NTUyNDcsIm5iZiI6MTcxNjU1NDk0NywicGF0aCI6Ii81OTY3NTA2OS8zMjk4MDQ4MTItOGNkN2Q2OWEtZWY4YS00ZjM4LTlhZjktZmZhZGY2ZWE0MGRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI0VDEyNDkwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIwZjcxZmU3NDU3NmFhOGNjZTNjZjlhM2JhYmU1OTNkOWEwNTcwYWJhNmQ3YjVmM2ZjZTFlZjMxZjUyNTE3ZWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.JBpq6zO4U-PMFJMzal_REDUm9NQYX-gJ0YF0Uo4n-50)

用LLAMA-factory训练了，一切正常

**Comment by leonunix** [comment]
{'loss': 3.3183, 'grad_norm': nan, 'learning_rate': 1.707941929974381e-08, 'epoch': 0.0}                                                                                                                  
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 1.707941929974381e-07, 'epoch': 0.0}  
我这也是一样的问题。

**Comment by leonunix** [comment]
通过升级bitsandbytes到最新，问题解决
bitsandbytes             0.42.0

升级到
bitsandbytes             0.43.1

**Comment by aa200647963** [comment]
我也遇到了这个问题，升级了bitsandbytes之后并没有解决

**Comment by leonunix** [comment]
@aa200647963 
你试试用Instruct的tokenizer。

**Comment by mst272** [comment]
> 通过升级bitsandbytes到最新，问题解决 bitsandbytes 0.42.0
> 
> 升级到 bitsandbytes 0.43.1

work!
